pyanaconda/dbus/server.py

```diff
@@ -310,13 +310,26 @@ def _connect_signal(self, interface_name, signal_name):
         :param signal_name: a DBus signal name
         :return: a disconnecting callback
         """
-        callback = partial(self._emit_signal, interface_name, signal_name)
+        callback = self._find_emitter(interface_name, signal_name)
         signal = self._find_handler(interface_name, signal_name)
         signal.connect(callback)
 
         disconnect = partial(signal.disconnect, callback)
         self._registrations.append(disconnect)
 
+    def _find_emitter(self, interface_name, signal_name):
+        """Find an emitter of a DBus signal.
+
+        :param interface_name: a DBus interface name
+        :param signal_name: a DBus signal name
+        :return: a callback
+        """
+        if interface_name == "org.freedesktop.DBus.Properties":
+            if signal_name == "PropertiesChanged":
+                return partial(self._emit_properties_changed)
```

Comment:
I am surprised by partial with only one argument.

Reply:
The handler doesn't support weak references that are used in our implementation of the signal. It means that registration of `self._emit_properties_changed` as a callback of a signal will fail with an exception. This code is temporary and will be changed, so I have only updated the commit message.